### PR TITLE
iptables_state: fix broken query of `async_status` result

### LINKED ISSUE #2627: iptables_state seems broken on ansible/ansible devel branch
### Summary

See:
- https://dev.azure.com/ansible/b24bf3ca-6168-45d7-99e2-bf8029e67c87/_apis/build/builds/16166/logs/1615
- https://dev.azure.com/ansible/b24bf3ca-6168-45d7-99e2-bf8029e67c87/_apis/build/builds/16167/logs/1563

```

2021-05-26T07:41:32.0705450Z 03:47 [0;31mThe full traceback is:[0m
2021-05-26T07:41:32.0706158Z 03:47 [0;31mTraceback (most recent call last):[0m
2021-05-26T07:41:32.0706897Z 03:47 [0;31m  File "/root/ansible/lib/ansible/executor/task_executor.py", line 158, in run[0m
2021-05-26T07:41:32.0707540Z 03:47 [0;31m    res = self._execute()[0m
2021-05-26T07:41:32.0708228Z 03:47 [0;31m  File "/root/ansible/lib/ansible/executor/task_executor.py", line 588, in _execute[0m
2021-05-26T07:41:32.0708923Z 03:47 [0;31m    result = self._handler.run(task_vars=variables)[0m
2021-05-26T07:41:32.0709716Z 03:47 [0;31m  File "/root/ansible_collections/community/general/plugins/action/iptables_state.py", line 130, in run[0m
2021-05-26T07:41:32.0710714Z 03:47 [0;31m    result = merge_hash(result, self._async_result(async_status_args, task_vars, 0))[0m
2021-05-26T07:41:32.0711599Z 03:47 [0;31m  File "/root/ansible_collections/community/general/plugins/action/iptables_state.py", line 55, in _async_result[0m
2021-05-26T07:41:32.0712343Z 03:47 [0;31m    if async_result['finished'] == 1:[0m
2021-05-26T07:41:32.0712928Z 03:47 [0;31mKeyError: 'finished'[0m
2021-05-26T07:41:32.0713495Z 03:47 [0;31mfatal: [testhost]: FAILED! => {[0m
2021-05-26T07:41:32.0714159Z 03:47 [0;31m    "msg": "Unexpected failure during module execution.",[0m
2021-05-26T07:41:32.0714756Z 03:47 [0;31m    "stdout": ""[0m
2021-05-26T07:41:32.0715240Z 03:47 [0;31m}[0m
```

### Issue Type

Bug Report

### Component Name

iptables_state

### Ansible Version

devel

### Configuration

*

### OS / Environment

*

### Steps to Reproduce

*

### Expected Results

*

### Actual Results

*

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by quidame**
##### SUMMARY

Fixes #2627 

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

iptables_state

### CODE REVIEW

File: changelogs/fragments/2671-fix-broken-query-of-async_status-result.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+bugfixes:
+  - "iptables_state - fix a broken query of ``async_status`` result
```
Comment (felixfontein) [inline changelogs/fragments/2671-fix-broken-query-of-async_status-result.yml:3]:
```suggestion
  - "iptables_state - fix a broken query of ``async_status`` result
    with current ansible-core development version
```

**Review by aminvakil — APPROVED** [review]
LGTM

**Comment by felixfontein** [comment]
@quidame thanks for fixing this!
@aminvakil thanks for reviewing!

**Comment by quidame** [comment]
@felixfontein @aminvakil thanks for reviewing !

**Pull request merged**